File: docs/test/samples/csharp/UtilityLibraries/StringLibraryTests/UnitTest1.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using UtilityLibraries;

namespace StringLibraryTest
{
    [TestClass]
    public class UnitTest1
    {
        [TestMethod]
        public void TestStartsWithUpper()
        {
            // Tests that we expect to return true.
            string[] words = { "Alphabet", "Zebra", "ABC", "Αθήνα", "Москва" };
            foreach (var word in words)
            {
                bool result = word.StartsWithUpper();
                Assert.IsTrue(result,
                              $"Expected for '{word}': true; Actual: {result}");
            }
        }

        [TestMethod]
        public void TestDoesNotStartWithUpper()
        {
            // Tests that we expect to return false.
            string[] words = { "alphabet", "zebra", "abc", "αυτοκινητοβιομηχανία", "государство",
                               "1234", ".", ";", " " };
            foreach (var word in words)
            {
                bool result = word.StartsWithUpper();
                Assert.IsFalse(result,
                               $"Expected for '{word}': false; Actual: {result}");
            }
        }

        [TestMethod]
        public void DirectCallWithNullOrEmpty()
        {
            // Tests that we expect to return false.
            string[] words = { String.Empty, null };
            foreach (var word in words)
            {
                bool result = StringLibrary.StartsWithUpper(word);
                Assert.IsFalse(result,
                               $"Expected for '{(word == null ? "<null>" : word)}': " +
                               $"false; Actual: {result}");

                //result = StringLibrary.StartsWithLower(word);
                //Assert.IsFalse(result,
                //               $"Expected for '{(word == null ? "<null>" : word)}': " +
                //               $"false; Actual: {result}");
            }
        }

        //[TestMethod]
        //public void TestStartsWithLower()
        //{
        //    // Tests that we expect to return false.
        //    string[] words = { "alphabet", "zebra", "abc", "αυτοκινητοβιομηχανία", "государство" };
        //    foreach (var word in words)
        //    {
        //        bool result = word.StartsWithLower();
        //        Assert.IsTrue(result,
        //                      $"Expected for '{word}': true; Actual: {result}");
        //    }
        //}

        //[TestMethod]
        //public void TestDoesNotStartWithLower()
        //{
        //    // Tests that we expect to return true.
        //    string[] words = { "Alphabet", "Zebra", "ABC", "Αθήνα", "Москва",
        //                       "1234", ".", ";", " "};
        //    foreach (var word in words)
        //    {
        //        bool result = word.StartsWithLower();
        //        Assert.IsFalse(result,
        //                      $"Expected for '{word}': false; Actual: {result}");
        //    }
        //}

        //[TestMethod]
        //public void TestHasEmbeddedSpaces()
        //{
        //    // Tests that we expect to return true.
        //    string[] phrases = { "one car", "Name\u0009Description",
        //                         "Line1\nLine2", "Line3\u000ALine4",
        //                         "Line5\u000BLine6", "Line7\u000CLine8",
        //                         "Line0009\u000DLine10", "word1\u00A0word2" };
        //    foreach (var phrase in phrases)
        //    {
        //        bool result = phrase.HasEmbeddedSpaces();
        //        Assert.IsTrue(result,
        //                      $"Expected for '{phrase}': true; Actual: {result}");
        //    }
        //}
    }
}

File: docs/test/samples/visual-basic/UtilityLibraries/StringLibraryTests/UnitTest1.vb
Imports Microsoft.VisualStudio.TestTools.UnitTesting
Imports UtilityLibraries

Namespace StringLibraryTests
    <TestClass>
    Public Class UnitTest1
        <TestMethod>
        Sub TestStartsWithUpper()
            ' Tests that we expect to return true.
            Dim words() As String = {"Alphabet", "Zebra", "ABC", "Αθήνα", "Москва"}
            For Each word In words
                Dim result As Boolean = word.StartsWithUpper()
                Assert.IsTrue(result,
                              $"Expected for '{word}': true; Actual: {result}")
            Next
        End Sub

        <TestMethod>
        Sub TestDoesNotStartWithUpper()
            ' Tests that we expect to return false.
            Dim words() As String = {"alphabet", "zebra", "abc", "αυτοκινητοβιομηχανία", "государство",
                               "1234", ".", ";", " "}
            For Each word In words
                Dim result As Boolean = word.StartsWithUpper()
                Assert.IsFalse(result,
                               $"Expected for '{word}': false; Actual: {result}")
            Next
        End Sub

        <TestMethod>
        Sub DirectCallWithNullOrEmpty()
            ' Tests that we expect to return false.
            Dim words() As String = {String.Empty, Nothing}
            For Each word In words
                Dim result As Boolean = StringLibrary.StartsWithUpper(word)
                Assert.IsFalse(result,
                               $"Expected for '{If(word Is Nothing, " < null > ", word)}': " +
                               $"false; Actual: {result}")

                'result = StringLibrary.StartsWithLower(word)
                'Assert.IsFalse(result,
                '               $"Expected for {If(word Is Nothing, " < null > ", word)}': " +
                '               $"false; Actual: {result}")
            Next
        End Sub

        '<TestMethod>
        'Sub TestStartsWithLower()
        '    ' Tests that we expect to return true.
        '    Dim words() As String = {"alphabet", "zebra", "abc", "αυτοκινητοβιομηχανία", "государство"}
        '    For Each word In words
        '        Dim result As Boolean = word.StartsWithLower()
        '        Assert.IsTrue(result,
        '                      $"Expected for '{word}': true; Actual: {result}")
        '    Next
        'End Sub

        '<TestMethod>
        'Public Sub TestDoesNotStartWithLower()
        '    ' Tests that we expect to return false.
        '    Dim words() As String = {"Alphabet", "Zebra", "ABC", "Αθήνα", "Москва",
        '                       "1234", ".", ";", " "}
        '    For Each word In words
        '        Dim result As Boolean = word.StartsWithLower()
        '        Assert.IsFalse(result,
        '                       $"Expected for '{word}': false; Actual: {result}")
        '    Next
        'End Sub

        '<TestMethod>
        'Public Sub TestHasEmbeddedSpaces()
        '    ' Tests that we expect to return true.
        '    Dim phrases() As String = {"one car", "Name" + vbTab + "Description",
        '                               "Line1" + vbCrLf + "Line2", "Line3" + vbLf + "Line4",
        '                               "Line5" + ChrW(&HB) + "Line6", "Line7" + ChrW(&HC) + "Line8",
        '                               "Line0009" + vbCr + "Line10", "word1" + ChrW(&HA0) + "word2"}
        '    For Each phrase In phrases
        '        Dim result As Boolean = phrase.HasEmbeddedSpaces()
        '        Assert.IsTrue(result,
        '                      $"Expected for '{phrase}': true; Actual: {result}")
        '    Next
        'End Sub
    End Class
End Namespace

